wrangler-service/src/main/java/io/cdap/wrangler/service/directive/RecipeHandler.java

```diff
@@ -147,4 +147,52 @@ public void deleteRecipe(HttpServiceRequest request, HttpServiceResponder respon
       responder.sendStatus(HttpURLConnection.HTTP_OK);
     });
   }
+
+  @PUT
+  @TransactionPolicy(value = TransactionControl.EXPLICIT)
+  @Path("v2/contexts/{context}/recipes/id/{recipe-id}")
+  public void updateRecipe(HttpServiceRequest request, HttpServiceResponder responder,
+                           @PathParam("context") String namespace,
+                           @PathParam("recipe-id") String recipeIdString) {
+    respond(responder, namespace, ns-> {
+      RecipeId recipeId = RecipeId.builder(ns).setRecipeId(recipeIdString).build();
+      // get recipe from request
+      Recipe recipe = GSON.fromJson(
+        StandardCharsets.UTF_8.decode(request.getContent()).toString(), Recipe.class);
+      // validate recipe
+      validateRecipeRequest(recipe);
```

Comment:
What about validating the case where recipe id passed as query param is not the same id of recipe passed in body of request?

Reply:
what we are doing right now is - we are ignoring the recipeId passed in the body of request and only using the recipe id passed as query param. Ideally, we should be creating a new class which only contains the fields that can be edited. Now thinking about it, I feel it's better we create a new object for Recipe edit request to avoid any confusion.